controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -510,6 +510,26 @@ func (r *BareMetalHostReconciler) actionUnmanaged(prov provisioner.Provisioner,
 	return actionContinue{unmanagedRetryDelay}
 }
 
+func (r *BareMetalHostReconciler) actionDetached(prov provisioner.Provisioner, info *reconcileInfo) actionResult {
+	if info.host.OperationalStatus() != metal3v1alpha1.OperationalStatusDetached {
```

Comment:
I don't actually see any harm in dropping this condition. We should always be able to call Detach() idempotently.

Reply:
Done!